### PR TITLE
[Core] Distance to path process

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

This adds a process able to compute the distance to a modelpart constituted of lines:

image

Depends on #10413

MPI not working for now, but I have higher order priorities right now, and I will fix it later.

**🆕 Changelog**

- [Initial commit](https://github.com/KratosMultiphysics/Kratos/commit/cffa08615cfe5394510fbf4e973bc7f7a7855c21)
- [Expose to python](https://github.com/KratosMultiphysics/Kratos/commit/95f54f540a2d075effd1fb5099a5166ad92356cd)
- [Test](https://github.com/KratosMultiphysics/Kratos/commit/9e4c0a9df43b0f08a58890181e794076903bcda6)

### CODE REVIEW

**Comment by ddiezrod** [comment]
Does this work in MPI? I have the impression that this will fail if you have nodes in one partition from the distance model part but you do not have any from path model part (this will be very common)

**Comment by loumalouomega** [comment]
> Does this work in MPI? I have the impression that this will fail if you have nodes in one partition from the distance model part but you do not have any from path model part (this will be very common)

Nope, it is in the description. A priori "I copy" the path model part in all partitions, but in my tests didn't work

**Comment by ddiezrod** [comment]
> > Does this work in MPI? I have the impression that this will fail if you have nodes in one partition from the distance model part but you do not have any from path model part (this will be very common)
> 
> Nope, it is in the description. A priori "I copy" the path model part in all partitions, but in my tests didn't work

Ok sorry, I missed that part.

File: kratos/processes/calculate_distance_to_path_process.cpp [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+        KRATOS_ERROR << "The model part is empty. Please check the model part provided" << std::endl;
+    }
+
+    // Now we create the vector of geometries
+    const unsigned int size_vector = r_elements_array.size();
+    if (r_distance_model_part.IsDistributed()) {
+        r_data_comm.SumAll(size_vector);
+    }
+    std::vector<Geometry<NodeType>::Pointer> geometries_vector(size_vector);
+    if (r_distance_model_part.IsDistributed()) {
```
Comment (loumalouomega) [inline kratos/processes/calculate_distance_to_path_process.cpp:87]: See @ddiezrod

**Comment by miguelmaso** [comment]
> This adds a process able to compute the distance to a modelpart constituted of lines:

very nice! Some time ago I did a dirty implementation of this for absorbing boundary conditions. I will use this process

**Comment by loumalouomega** [comment]
It would be fine if we advance the merge of https://github.com/KratosMultiphysics/Kratos/pull/10413, so we can proceed with this one

**Comment by loumalouomega** [comment]
Hello everyone

**Comment by loumalouomega** [comment]
> Hello everyone, specially @KratosMultiphysics/altair

👋

**Comment by loumalouomega** [comment]
> > Hello everyone, specially @KratosMultiphysics/altair
> 
> 👋

👋👋👋👋👋

**Comment by loumalouomega** [comment]
Done @ddiezrod

**Review by ddiezrod — DISMISSED** [review]
Looks good, approving!

**Review by ddiezrod — APPROVED** [review]

File: kratos/processes/calculate_distance_to_path_process.h [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+     * @return The Distance computation type
+     * @todo Add C++ test
+     */
+    static DistanceComputed FastMinimalDistanceOnLineWithRadius(
+        double& rDistance,
+        const Geometry<NodeType>& rSegment,
+        const Point& rPoint,
+        const double Radius,
+        const double Tolerance = 1.0e-9
+        );
```
Comment (rubenzorrilla) [inline kratos/processes/calculate_distance_to_path_process.h:245]: Why adding a `static` method to the process? This must be in a separated util IMO.

File: kratos/processes/calculate_distance_to_path_process.h [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+///@{
+
+/**
+ * @class PointGeometry
+ * @ingroup KratosCore
+ * @brief Custom Point container to be used by the search
+ * @details It stores the pointer of a certain geometry
+ * @author Vicente Mataix Ferrandiz
+ */
+class PointGeometry
```
Comment (rubenzorrilla) [inline kratos/processes/calculate_distance_to_path_process.h:63]: Why you need this?

File: kratos/processes/calculate_distance_to_path_process.h [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+ */
+enum class DistanceComputed
+{
+    ERROR,
+    NO_RADIUS,
+    RADIUS_PROJECTED,
+    RADIUS_NOT_PROJECTED_OUTSIDE,
+    RADIUS_NOT_PROJECTED_INSIDE
+};
+
```
Comment (rubenzorrilla) [inline kratos/processes/calculate_distance_to_path_process.h:47]: Same comment as for the auxiliary PR you did for this. I don't like the idea of having these global enum definitions. Even worse if the unique purpose is for the `static` method you are creating inside the class.

**Comment by loumalouomega** [comment]
@rubenzorrilla tomorrow I will take into account your comments in a new PR.

File: kratos/processes/calculate_distance_to_path_process.h (hunk 3 above)
Comment (loumalouomega) [reply kratos/processes/calculate_distance_to_path_process.h:63]: For the KDTree

**Comment by rubenzorrilla** [comment]
> @rubenzorrilla tomorrow I will take into account your comments in a new PR.

@loumalouomega  any advance in this regard?

**Pull request merged**